src/corehost/corehost.cpp

```diff
@@ -161,32 +246,79 @@ bool resolve_fxr_path(const pal::string_t& host_path, const pal::string_t& app_r
 
     if (max_ver == fx_ver_t())
     {
-        trace::error(_X("A fatal error occurred, the folder [%s] does not contain any version-numbered child folders"), fxr_dir.c_str());
+        trace::error(_X("A fatal error occurred, the folder [%s] does not contain any version-numbered child folders"), fxr_root.c_str());
         return false;
     }
 
     pal::string_t max_ver_str = max_ver.as_str();
-    append_path(&fxr_dir, max_ver_str.c_str());
-    trace::info(_X("Detected latest fxr version=[%s]..."), fxr_dir.c_str());
+    append_path(&fxr_root, max_ver_str.c_str());
+    trace::info(_X("Detected latest fxr version=[%s]..."), fxr_root.c_str());
 
-    if (library_exists_in_dir(fxr_dir, LIBFXR_NAME, out_fxr_path))
+    if (library_exists_in_dir(fxr_root, LIBFXR_NAME, out_fxr_path))
     {
-        trace::info(_X("Resolved fxr [%s]..."), out_fxr_path ->c_str());
+        trace::info(_X("Resolved fxr [%s]..."), out_fxr_path->c_str());
         return true;
     }
 
-    trace::error(_X("A fatal error occurred, the required library %s could not be found in [%s]"), LIBFXR_NAME, fxr_dir.c_str());
+    trace::error(_X("A fatal error occurred, the required library %s could not be found in [%s]"), LIBFXR_NAME, fxr_root.c_str());
 
     return false;
 }
 
-int run(const int argc, const pal::char_t* argv[])
+#if defined(CURHOST_LIB)
+
+int get_com_activation_delegate(pal::string_t *app_path, com_activation_fn *delegate)
+{
+    pal::string_t host_path;
+    if (!pal::get_own_module_path(&host_path) || !pal::realpath(&host_path))
+    {
+        trace::error(_X("Failed to resolve full path of the current host module [%s]"), host_path.c_str());
+        return StatusCode::CoreHostCurHostFindFailure;
+    }
+
+    pal::string_t dotnet_root;
+    pal::string_t fxr_path;
+    if (!resolve_fxr_path(host_path, &dotnet_root, &fxr_path))
+    {
+        return StatusCode::CoreHostLibMissingFailure;
+    }
+
+    // Load library
+    pal::dll_t fxr;
+    if (!pal::load_library(&fxr_path, &fxr))
+    {
+        trace::error(_X("The library %s was found, but loading it from %s failed"), LIBFXR_NAME, fxr_path.c_str());
+        trace::error(_X("  - Installing .NET Core prerequisites might help resolve this problem."));
+        trace::error(_X("     %s"), DOTNET_CORE_INSTALL_PREREQUISITES_URL);
+        return StatusCode::CoreHostLibLoadFailure;
+    }
+
+    // Leak fxr
+
+    auto entry_del = (hostfxr_get_com_activation_delegate_fn)pal::get_symbol(fxr, "hostfxr_get_com_activation_delegate");
```

Comment:
Nit: please rename the variable to `hostfxr_get_com_activation_delegate`, it makes the code much easier to read since it looks exactly like if there's a direct call (and no `GetProcAddress` magic).

Reply:
But it isn't a direct call. It is a call based on some very important logic (i.e. the call is depending on another library). Obscuring that doesn't help readability of code, it loses the context of a loose dependency. I will change it if that is the style, but I think the proposed argument for changing it is flawed.

Reply:
Hmm - OK - I didn't see it that way.
What I found confusing was the last line of this function - call to `entry_del`:
* I didn't find this as a function in the source code (I should have thought about function pointers, but still)
* I was trying to make sense of what it does from the name - and it read as "delete entry" or something along those lines.

Using the name of the export would have made it very clear to me that this is call to hostfxr and what the entry point is.
So I'm OKish to go with a name which doesn't necessarily match the export exactly, but please do rename this to something more descriptive.